### PR TITLE
Add dependabot

### PR DESCRIPTION
**Pull request by ssteinbach**
Adds a dependabot.yml to opentimelineio.

Dependabot is a GitHub service that opens Pull requests to update the version of dependencies in the library.  For more information:
https://docs.github.com/en/code-security/supply-chain-security/enabling-and-disabling-version-updates#enabling-github-dependabot-version-updates

### CODE REVIEW

**Comment by jhodges10** [comment]
Good call making it Monday's only @ssteinbach, all too often dependabot gets incredibly annoying because of incessant notifications.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+version: 2
+updates:
+
+  # Maintain dependencies for GitHub Actions
+  # - package-ecosystem: "github-actions"
+  #   directory: "/"
+  #   schedule:
+  #     interval: "daily"
+
+  # Maintain dependencies for npm
```
Comment (reinecke) [inline .github/dependabot.yml:10]: This should say `pip` instead of `npm`.

**Review by reinecke — COMMENTED** [review]
Just one copy/paste issue in a comment. This is a good add though, especially for our submodules.

File: .github/dependabot.yml (hunk 1 above)
Comment (ssteinbach) [inline .github/dependabot.yml:10]:
```suggestion
  # Maintain dependencies for pip
```

**Review by reinecke — APPROVED** [review]

**Pull request merged**